compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml

```diff
@@ -196,16 +196,25 @@ class HasFromAnyContractKey t k | t -> k where
   -- | HIDE
   _fromAnyContractKey : proxy t -> Any -> Optional k
 
+-- | Constraint satisfied by choices from templates with key.
+type ChoiceByKey t k c r =
+  ( TemplateKey t k
+  , HasExerciseByKey t k c r
+  )
```

Comment:
Maybe this constraint tuple should also include the other classes from `Choice t c r`,

```suggestion
type ChoiceByKey t k c r =
  ( TemplateKey t k
  , HasExercise t c r
  , HasToAnyChoice t c r
  , HasFromAnyChoice t c r
  , HasExerciseByKey t k c r
  )
```

Reply:
why not 
```suggestion
type ChoiceByKey t k c r =
  ( Choice t c r
  , HasExerciseByKey t k c r
  )
```
?

Reply:
You would lose the things defined in `TemplateKey t k` that are not in `Template t`, since `Choice t c r` only has `Template t`.

Reply:
But I kind of agree with Moritz now that maybe it's better not to define this constraint tuple for now.